_resources/port1.0/compilers/gcc_compilers.tcl

```diff
@@ -8,9 +8,10 @@ if {${os.major} >= 10 || [option os.platform] ne "darwin"} {
     lappend compilers macports-gcc-14 macports-gcc-13 macports-gcc-12 macports-gcc-11 macports-gcc-10
 }
 
-# GCC 9 and older only on OSX10.10 and older
+# GCC 14, GCC 9 and older only on OSX10.10 and older
```

Comment:
we are also going to use gcc10 on older systems as a fallback compiler. I can’t see enough code in this PR right now to make sure that is set right, but on older systems it needs to be in the mix to allow that

Reply:
As I understand from previous discussion, gcc10 is only used on older systems via gcc10-bootstrap in older to compile gcc14.
For the non-bootstrap compiler, the list of newer compilers is gcc9 and gcc14, right?

Reply:
I suggested gcc9, but @barracuda156 argues hard for gcc10 instead, so I aquiesed.

Reply:
OK, great! Very glad to hear that we won't have to implement gcc9 support.

Currently we have gcc10-bootstrap, I'm guessing getting the non-bootstrap out of that won't be too difficult.

Reply:
I hope this whole mechanism is getting pretty clear to all you folks by now, as we’re going to count on you running with it/ fixing it and fielding tickets for the next few years

Reply:
> we still have to implement gcc9 building, as we need libgcc9 as it contributes unique libraries.

This is just out of curiosity but is there an example of such a unique library and software that still uses it? As I understand, such a library would be a gcc extension rather than something standard.

> I hope this whole mechanism is getting pretty clear to all you folks by now, as we’re going to count on you running with it/ fixing it and fielding tickets for the next few years

Yeah, starting to get the hang of things :)

Reply:
Can you please @-mention Chris here (I don't know who that is), perhaps he remembers what that unique need was?

Reply:
Ah, I see. How do I figure out the list of packages from the zip file?

I tried looking at the structure:

```bash
$ tree libgcc9-9.5.0_2.darwin_10.i386 
libgcc9-9.5.0_2.darwin_10.i386
├── +COMMENT
├── +CONTENTS
├── +DESC
├── opt
│   └── local
│       └── lib
│           └── libgcc
│               ├── libgcc_ext.10.4.dylib
│               └── libgcc_ext.10.5.dylib
├── +PORTFILE
└── +STATE
```

But it just seems to be libgcc itself, not a list of things that need it.

Reply:
It's all of libgcc9, not something that has a need for it. It's very hard to figure out what has to be investigated/tested without a concrete example.

Reply:
Still, unique needed libs, looked for by gcc9 and earlier compilers.

Reply:
So your worry is basically that setting libgcc14 as the default could break gcc7.

Since we don't know the status of gcc8-13, would it be considered blockers if they turn out to already not work?

Is there a way to list things that depends on specific GCC versions like say GCC8 or is that not relevant to what you are thinking about?

Reply:
when libgcc9 is confirmed to build on all systems and install it’s unique library, there will be no worries. 

so libgcc9 (and therefore gcc9, which builds during the build of libgcc9) need to build.

@barracuda156 wants gcc10 to be the interim compiler between gcc7 and gcc14, so that needs to build on all systems too. Should be easy enough, as we long ago fixed gcc10-bootstrap to build on all systems.

gcc8, 11, 12, and 13 can be ignored, as they are not going to be in the fallback compiler list (at least to start), but some of them may build, may be desired by someone, or may be actually needed for something we don’t know about yet. The libgcc setup will support any/all of them.

Reply:
As long a macports doesn't provide binaries for older Macs having more than the absolute minimum is very understanding, each version of GCC adds about a full day to the installation process. Just installing bash takes about a week on a fresh setup because of the build dependencies. Are there any way this could be addressed, anything we can do to help?

@barracuda156 since gcc9 is being insisted on do you think we can drop gcc10?

Reply:
Ryan won’t allow anyone outside his umbrella to build and supply software that macports will distribute due to security concers. Ryan needs the long screwdriver required to get the heat sink off his G5 so he can change the power supply. That is the main reason there is no 10.5 ppc builder. He has no hardware to add a Tiger builder, and might accept a donation of such from somewhere.

Reply:
Well I don't care so much about 10.5 since I can't run it on my machine. The G5 should be able to run 10.4.2+ though so he technically should have the hardware for it, but maybe dual booting it isn't feasible.

Roughly where in the world is Ryan located?

Reply:
Ryan is in Texas.

Reply:
Then this might be an option for us: https://www.ebay.com/itm/276590374857?mkcid=16&mkevt=1&mkrid=711-127632-2357-0&ssspo=G7qszA03Tqm&sssrc=2047675&ssuid=c4ie_iqvqzo&widget_ver=artemis&media=COPY

Reply:
Ryan won’t be looking here…. this would be something for the macports-users mailing list

Reply:
Would you be willing to point @ryandesign here, or is he simply unwilling to visit GitHub? 

I'm not willing to subscribe to some mailing list, which I see is a requirement to post there, I can barely manage my inbox as is.

Reply:
You have to register once to post, to prevent spammers, but you don’t need to subscribe to the mailing list. 

But ok. 

@ryandesign — folks here would like to send you ppc hardware if you might be able to set up a 10.4 and 10.5 builder. They might also send you the long screwdriver you need to fix your existing machine. Are you able to accommodate thrm?

Reply:
> we are also going to use gcc10 on older systems as a fallback compiler. I can’t see enough code in this PR right now to make sure that is set right, but on older systems it needs to be in the mix to allow that

I've pushed a commit addressing the feedback. I haven't tested it yet because I'm still compiling (lib)gcc14.

Before I test it, can you please have a look to see if the changes make sense and actually address your feedback?

Reply:
very very close.

gcc10 still needs it’s platforms adjusted.

that’s all I could spot.

Reply:
I've removed the `darwin >= 10` restriction from gcc10 but also noticed that it has a lot fewer patches than gcc10-bootstrap. If it fails to build, we'll probably simply need to copy the patches over.